lib/datasource/nuget/index.js

```diff
@@ -201,3 +208,19 @@ async function getQueryUrlForV3Feed(url) {
     return null;
   }
 }
+
+function authenticatedGot(url, options) {
+  let finalOptions = options;
+  const { host } = urlApi.parse(url);
+  const hostRule = hostRules.find({ platform: 'nuget', host });
+  if (hostRule && hostRule.username && hostRule.password) {
+    const auth = Buffer.from(
+      `${hostRule.username}:${hostRule.password}`
+    ).toString('base64');
+    finalOptions = {
+      ...finalOptions,
+      headers: { Authorization: `Basic ${auth}` },
```

Comment:
Careful because if the existing `finalOptions` has any `headers` then you'll overwrite them. You either want to use some type of nested merge function that combines sub-objects, or just take a few extract lines to do like this:

```js
finalOptions.headers = finalOptions.headers || {};
finalOptions.headers.Authorization = `Basic ${auth}`;
```

Reply:
Also, this function is pretty close to something I've been thinking to add to the `lib/util/got/*` library of modules :)

Reply:
Thanks for the hint, will update!

I just found the stuff at `lib/util/got`, though I'm not sure we can do it 100% transparently because the `platform` needs to go into the `hostRules` lookup logic...

Reply:
You could perhaps do part of it transparently if the hostRule is set using the *host* only and not also platform. But it's been a while since I played with it. I'm totally fine keeping it within this file and one day I'll see if I can migrate all the auth stuff "down" the stack to make it semi-invisible as part of our got wrappers.

Reply:
TBH, I’d prefer to keep it local for now - I’m in a love/hate relationship with JavaScript and all it’s magic patterns/libs :)

Reply:
The header merging problem is addressed with c12ea6c